python/ollama.py

```diff
@@ -0,0 +1,146 @@
+import weechat
+import json
+
+"""
+Ollama Bot for WeeChat (Non-blocking version)
+
+This script automatically responds to mentions in channels and private messages using an Ollama LLM running locally.
+
+Features:
+- Responds to mentions in channels.
+- Can respond to private messages if enabled.
+- Allows manual queries using the /ollama command.
+- Configurable via WeeChat /set commands.
+- Uses hook_url for non-blocking HTTP requests.
```

Comment:
This comment says it uses `hook_url`, but `hook_process_hashtable` is used instead.

Using hook_url has 2 advantages:

- it does not rely on an external curl command (which may not be available), but it still uses curl internally (linked with weechat)
- it uses a thread instead of a process, which is more lightweight.